lib/transfer.cpp

```diff
@@ -455,7 +455,7 @@ namespace quda {
 
       if (use_gpu) {
         if (out.Location() == QUDA_CPU_FIELD_LOCATION) output = coarse_tmp_d;
-        if (in.Location() == QUDA_CPU_FIELD_LOCATION || in.GammaBasis() != V->GammaBasis())
+        if (in.Location() == QUDA_CPU_FIELD_LOCATION || in.GammaBasis() != V->GammaBasis() || in.FieldOrder() != QUDA_FLOAT2_FIELD_ORDER)
```

Comment:
Remind me: why did you need this extra check for FLOAT2-ordered fields?

Reply:
The context here was restricting Wilson-type near-null vectors from level 0 -> level 1; the fine `B` vectors were in `FLOAT4` order. (I'm sure this was an issue before, but I don't know if the restriction workflow had been tested in a long while.)

Reply:
Ah gotcha.  Yes, this makes sense.

it’s an issue for the B vectors because they are in a relativistic basis so don’t trigger the copy unlike a general residual vector which will be in the non relativistic basis.